labgrid/util/managedfile.py

```diff
@@ -53,12 +48,41 @@ def sync_to_resource(self):
         if isinstance(self.resource, NetworkResource):
             host = self.resource.host
             conn = sshmanager.open(host)
+
+            if self._on_nfs(conn) == True:
+                return # nothing to do
+
+            self.rpath = "/tmp/labgrid-{user}/{hash}/".format(
+                user=get_user(), hash=self.get_hash()
+            )
             conn.run_check("mkdir -p {}".format(self.rpath))
+
             conn.put_file(
                 self.local_path,
                 "{}{}".format(self.rpath, os.path.basename(self.local_path))
             )
 
+    def _on_nfs(self, conn):
+        if self._on_nfs_cached is not None:
+            return self._on_nfs_cached
+
+        if not self.detect_nfs:
+            return False
+
+        fmt="inode=%i,size=%s,birth=%W,modified=%Y"
+        remote = conn.run("stat --format '{}' {}".format(fmt, self.local_path),
+                          decodeerrors="backslashreplace")
+        local = subprocess.run(["stat", "--format", fmt, self.local_path],
+                               stdout=subprocess.PIPE)
+        localout = local.stdout.decode("utf-8", "backslashreplace").split('\n')
+        localout.pop() # remove trailing empty element
+
+        self._on_nfs_cached = remote[2] == local.returncode == 0 and remote[0] == localout
```

Comment:
Please use a debug logger for separate error cases.

Reply:
A different debug message for each boolean check, you mean? I've done so now.